openssl/src/pkey.rs

```diff
@@ -621,6 +777,45 @@ impl PKey<Public> {
         PKey<Public>,
         ffi::d2i_PUBKEY
     }
+
+    #[cfg(ossl111)]
+    fn import_raw_public_key(typ: c_int, key: &[u8]) -> Result<PKey<Public>, ErrorStack> {
```

Comment:
I would just make this public and take a Type, and then remove the other constructors that delegate to it.